java/arcs/android/devtools/StoreMessage.kt

```diff
@@ -1,156 +1,44 @@
-                    )
-                }
-                is CrdtSet.Operation.Clear<*> -> {
-                    list.add(
-                        JsonValue.JsonObject(
-                            TYPE to JsonValue.JsonString(CLEAR_TYPE),
-                            ACTOR to JsonValue.JsonString(op.actor),
-                            CLOCK to op.clock.toJson()
-                        )
-                    )
-                }
-                is CrdtSet.Operation.Remove<*> -> {
-                    list.add(
-                        JsonValue.JsonObject(
-                            TYPE to JsonValue.JsonString(REMOVE_TYPE),
-                            REMOVED to getValue(op.removed),
-                            ACTOR to JsonValue.JsonString(op.actor),
-                            CLOCK to op.clock.toJson()
-                        )
-                    )
-                }
-                is CrdtSet.Operation.FastForward<*> -> {
-                    list.add(
-                        JsonValue.JsonObject(
-                            TYPE to JsonValue.JsonString(FAST_FORWARD_TYPE),
-                            ADDED to getAddedListValue(op.added),
-                            REMOVED to getRemovedListValue(op.removed),
-                            OLD_CLOCK to JsonValue.JsonString(op.oldClock.toString()),
-                            CLOCK to op.clock.toJson()
-                        )
-                    )
-                }
-            }
-        }
-        return list
-    }
+interface StoreMessage : DevToolsMessage {
 
     /**
      * Turn the clock [VersionMap] into a [JsonValue.JsonObject].
      */
-    private fun VersionMap.toJson(): JsonValue<*> {
+    fun VersionMap.toJson(): JsonValue<*> {
         val map = mutableMapOf<String, JsonValue<*>>()
         actors.forEach {
             map.put(it, JsonValue.JsonNumber(this[it].toDouble()))
         }
         return JsonValue.JsonObject(map)
     }
 
-    /**
-     * Return the items in the removed list as a JsonArray
-     */
-    private fun getRemovedListValue(list: MutableList<out Referencable>): JsonValue.JsonArray {
-        val array = mutableListOf<JsonValue<*>>()
-        list.forEach {
-            array.add(getValue(it))
-        }
-        return JsonValue.JsonArray(array)
-    }
-
-    /**
-     * Return the items in the added list as a JsonArray
-     */
-    private fun getAddedListValue(
-        list: MutableList<out CrdtSet.DataValue<out Referencable>>
-    ): JsonValue.JsonArray {
-        val array = mutableListOf<JsonValue<*>>()
-        list.forEach {
-            array.add(getValue(it.value))
-        }
-        return JsonValue.JsonArray(array)
-    }
-
     /**
      * Get the primitive value of a [Referencable].
      */
-    private fun getValue(value: Referencable?): JsonValue<*> {
+    fun getValue(value: Referencable?): JsonValue<*> {
         when (value) {
+            is Reference -> {
+                return JsonValue.JsonObject(
```

Comment:
Nit: This when could also be turned into an expression like in some of the other comments on the PR.